### PR TITLE
Add support for TypeScript 4

### PR DESCRIPTION
**Pull request by SergioMorchon**
This PR includes small changes required to prepare for TypeScript 4.
It's currently in beta, so **hold on** this merge.

### CODE REVIEW

**Comment by Hexxeh** [comment]
We've pulled the latest final TypeScript 4.0 build into the 4.2/5.0 releases. Thanks for trying this out early though! :)

**Comment by Hexxeh** [comment]
I take that back... TS 4.0 didn't get included in the latest round, I thought it had.

**Comment by SergioMorchon** [comment]
No worries, I did it as an exercise to advance any issues; I'll try to update this branch

File: src/__snapshots__/compile.test.ts.snap [hunk 1]
```diff
@@ -42,7 +42,7 @@ PERFORMANCE OF THIS SOFTWARE.
...
 "\\"use strict\\"
-var o,t=(o=function(t,n){return(o=Object.setPrototypeOf||{__proto__:[]}instanceof Array&&function(o,t){o.__proto__=t}||function(o,t){for(var n in t)t.hasOwnProperty(n)&&(o[n]=t[n])})(t,n)},function(t,n){function r(){this.constructor=t}o(t,n),t.prototype=null===n?Object.create(n):(r.prototype=n.prototype,new r)});(new(function(o){function n(){return null!==o&&o.apply(this,arguments)||this}return t(n,o),n.prototype.doBar=function(){console.log(\\"foo\\")},n}(function(){function o(){}return o.prototype.doFoo=function(){},o}()))).doBar()
+var t,o=(t=function(o,n){return(t=Object.setPrototypeOf||{__proto__:[]}instanceof Array&&function(t,o){t.__proto__=o}||function(t,o){for(var n in o)Object.prototype.hasOwnProperty.call(o,n)&&(t[n]=o[n])})(o,n)},function(o,n){function r(){this.constructor=o}t(o,n),o.prototype=null===n?Object.create(n):(r.prototype=n.prototype,new r)});(new(function(t){function n(){return null!==t&&t.apply(this,arguments)||this}return o(n,t),n.prototype.doBar=function(){console.log(\\"foo\\")},n}(function(){function t(){}return t.prototype.doFoo=function(){},t}()))).doBar()
```
Comment (SergioMorchon) [inline src/__snapshots__/compile.test.ts.snap:45]: Looks like the same code, jsut swapping the `o` and `t` variable names

File: yarn.lock [hunk 2]
```diff
@@ -5574,10 +5574,10 @@ tr46@^2.0.0:
   dependencies:
     punycode "^2.1.1"
 
-ts-jest@^26.2.0:
-  version "26.2.0"
-  resolved "https://registry.yarnpkg.com/ts-jest/-/ts-jest-26.2.0.tgz#7ec22faceb05ee1467fdb5265d1b33c27441f163"
-  integrity sha512-9+y2qwzXdAImgLSYLXAb/Rhq9+K4rbt0417b8ai987V60g2uoNWBBmMkYgutI7D8Zhu+IbCSHbBtrHxB9d7xyA==
+ts-jest@^26.3.0:
```
Comment (SergioMorchon) [inline yarn.lock:5577]: I've updated this, as it supports TS4, while previous version didn't, so it solves this warning:
> ts-jest[versions] (WARN) Version 4.0.2 of typescript installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=3.8.0 <4.0.0-0). Please do not report issues in ts-jest if you are using unsupported versions.

**Comment by SergioMorchon** [comment]
@Hexxeh do you have any plan to keep updated the TS version?
I'm interested in some new features to explore, like https://devblogs.microsoft.com/typescript/announcing-typescript-4-1/#template-literal-types (i.e. for defining event listeners and remove a lot of boilerplate).
And without this project using that version, I think that the type definitions I have in https://github.com/SergioMorchon/fitbit-sdk-types won't work

**Comment by Hexxeh** [comment]
We typically would update it with each SDK release. I normally go and update everything just before we start QA for a release. Happy to take this merged in though once it passes again if you want to get it in `master` early.

**Comment by SergioMorchon** [comment]
Nice! I've updated TypeScript to `4.1.3`

**Review by Hexxeh — APPROVED** [review]

**Pull request merged**